File: src/main/kotlin/FizzBuzz/java/Solution1.java
package FizzBuzz;

import java.util.ArrayList;
import java.util.List;

public class Solution1 {
    public List<String> fizzBuzz(int n) {
        List<String> answer = new ArrayList<>(n);
        for (int i = 1; i <= n; i++) {
            boolean divisibleBy3 = i % 3 == 0;
            boolean divisibleBy5 = i % 5 == 0;
            if (divisibleBy3 && divisibleBy5) {
                answer. add("FizzBuzz");
            } else if (divisibleBy3) {
                answer.add("Fizz");
            } else if (divisibleBy5) {
                answer.add("Buzz");
            } else {
                answer.add(String.valueOf(i));
            }
        }
        return answer;
    }
// Time complexity = O(n)
// Space complexity = 0(1)
}

File: src/main/kotlin/FizzBuzz/kotlin/Solution1.kt
package FizzBuzz.kotlin

class Solution1 {
    fun fizzBuzz(n: Int): List<String> {
        val answer: MutableList<String> = ArrayList(n)
        for (i in 1..n) {
            val divisibleBy3 = i % 3 == 0
            val divisibleBy5 = i % 5 == 0
            if (divisibleBy3 && divisibleBy5) {
                answer.add("FizzBuzz")
            } else if (divisibleBy3) {
                answer.add("Fizz")
            } else if (divisibleBy5) {
                answer.add("Buzz")
            } else {
                answer.add(i.toString())
            }
        }
        return answer
    } // Time complexity = O(n)
    // Space complexity = 0(1)
}